spec.bs

```diff
@@ -248,6 +248,88 @@ If using a centralized mechanism for fetching keys, user agents should have a re
 
 When using a registration process, it is recommended that user agents apply an expiration date to registration requests, to allow for the removal of deprecated or no longer active issuers.
 
+VOPRF Methods {#voprf-methods}
+====================================
+
+For Issue (corresponding to **BlindEvaluate** function of the [[!VOPRF]] protocol), the Request and Response are serialized as:
+
+```
+struct {
+  uint16 count;
+  ECPoint nonces[count]; // Encoded as X9.62 Uncompressed point.
+} IssueRequest;
+
+struct {
+  ECPoint blinded; // Encoded as X9.62 Uncompressed point.
+  ECPoint evaluated; // Encoded as X9.62 Uncompressed point.
+} SignedNonce;
+
+struct {
+  Scalar c;
+  Scalar u;
+  Scalar v;
+} DLEQProof;
+
+struct {
+  uint16 issued;
+  uint32 key_id = keyID;
+  SignedNonce signed[issued];
+  opaque proof<1..2^16-1>; // Length-prefixed form of DLEQProof.
+} IssueResponse;
+```
+
+For Redeem (corresponding to **Evaluate** function of the [[!VOPRF]] protocol), the Request and Response are serialized as:
+
+```
+struct {
+  uint32 key_id;
```

Comment:
missing: 
`uint16 token_length` (before key_id)
`uint16 client_data_length` (before client_data)

Reply:
In TLS syntax, the "type name<1..2^size-1>" construction indicates a length-prefix of the appropriate number of bytes.